react/src/DateRangePicker/DateRangePickerContext.tsx

```diff
@@ -278,7 +273,7 @@ const useProvideDateRangePicker = ({
   const handleCalendarDateChange = useCallback(
     (date: DateRangeValue) => {
       const zonedDateRange = date.map((d) =>
-        d ? zonedTimeToUtc(d, timeZone) : null,
+        d,
```

Comment:
This seems redundant. Could we use the passed in date instead?

```tsx
 const handleCalendarDateChange = useCallback(
    (date: DateRangeValue) => {
         setInternalValue(date)
         ....
    }. [])
```

Reply:
Yup, you're right. Sorry, your suggestion was what I did on Form but neglected to do here 😅

Reply:
done in [ed85384](https://github.com/opengovsg/design-system/pull/309/commits/ed8538418ec4ad1fbdf27d534a09fd1a9d1de140)